core/src/main/java/org/frankframework/pipes/JsonPipe.java

```diff
@@ -161,14 +172,23 @@ public void setVersion(String version) {
 
 	/**
 	 * When direction is JSON2XML, it wraps a root element around the converted message.
-	 * When direction is XML2JSON, it includes the name of the root element as a key in the converted message.
+	 * When direction is XML2JSON, it includes the name of the root element as a key in the converted message, thus preserving the structure of the original input message.
 	 * @ff.default TRUE when JSON2XML and FALSE when XML2JSON
 	 */
 	public void setAddXmlRootElement(boolean addXmlRootElement) {
 		this.addXmlRootElement = addXmlRootElement;
 	}
 
-	/** Format the output in easy legible way (currently only for XML) */
+	/**
+	 * When direction is JSON2XML, specifies the name of the root element when {@literal addXmlRootElement} is {@code true}.
```

Comment:
So the custom root name is ignored when `addXmlRootElement` is `false`?
Perhaps that should also be verified then in `configure()`? Not sure if that should also throw an error but I guess that it should?